Storage/REST-API.md

```diff
@@ -0,0 +1,185 @@
+# RESTful binding for Storage, RFC
+
+This document describes the RESTful API with CRUD functionality for Current Storage.
+
+### Bindings semantics
+`TODO: Describe storage field <-> endpoint bindings.`
+
+### Response schema
+
+The response body is always either a valid JSON or empty with `204 (No content)` error code.
+
+The generic format of a successful JSON response is:
+
+```
+{
+  "url": "https://api.example.com/...",
+  "url_*": "https://api.example.com/...",
+  ... more "url_*" fields ...
+  
+  [optional]
+  "meta": {
+    ...
+  },
+  
+  [required]
+  "data": {
+    ...
+  }
+
+}
+```
+
+The responses are strongly typed. The contents of the request URL define the schema for the response.
```

Comment:
I keep re-reading this and seems to me that the words "strongly typed" should be accompanied with a formal way to enforce it. It's not a written rule, I'm just expecting it here, subjectively. Since the schema (as far as I see) is in the prose of the spec, it may be better to just say that all responses adhere to the format(s) defined in this spec.